torchtext/experimental/datasets/sequence_tagging.py

```diff
@@ -27,16 +27,16 @@ def _setup_datasets(dataset_name,
                     root=".data",
                     vocabs=None,
                     data_select=("train", "valid", "test")):
-    if isinstance(data_select, str):
-        data_select = [data_select]
-    if not set(data_select).issubset(set(("train", "valid", "test"))):
-        raise TypeError("Given data selection {} is not supported!".format(data_select))
-
-    train, val, test = raw.DATASETS[dataset_name](root=root)
+    data_select = check_default_set(data_select, ('train', 'valid', 'test'))
+    if dataset_name == 'UDPOS':
+        train, val, test = raw.DATASETS[dataset_name](root=root)
+    elif dataset_name == 'CoNLL2000Chunking':
+        train, test = raw.DATASETS[dataset_name](root=root)
+        val = None
     raw_data = {
```

Comment:
You can just iterate over data_select instead of doing the "if train and 'train' in[...]"

Reply:
```
raw_iterator = {}
for name, data in zip(data_select, raw.DATAsets[dataset_name](root=root, data_select=data_select)):
    raw_iterator[name] = data
for name in data_select:
    raw_data[name] = list(raw_iterator[name])
```

Reply:
Done